### PR TITLE
http: backport new events

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/2655
https://redmine.openinfosecfoundation.org/issues/1776

Describe changes:
- backport of https://github.com/OISF/suricata/pull/3705

Makes https://github.com/OISF/suricata-verify/pull/24 pass
and https://github.com/OISF/suricata-verify/pull/54 as well

Replaces #4229 with better rebase

### CODE REVIEW

File: src/app-layer-htp.c [hunk 1]
```diff
@@ -488,6 +521,8 @@ struct {
 /*    { "Invalid authority port", HTTP_DECODER_EVENT_INVALID_AUTHORITY_PORT}, htp no longer returns this error */
     { "Request buffer over", HTTP_DECODER_EVENT_REQUEST_FIELD_TOO_LONG},
     { "Response buffer over", HTTP_DECODER_EVENT_RESPONSE_FIELD_TOO_LONG},
+    { "C-T multipart/byteranges in responses not supported", HTTP_DECODER_EVENT_RESPONSE_MULTIPART_BYTERANGES},
```
Comment (jlucovsky) [inline src/app-layer-htp.c]: Suggest `byte-ranges` instead

Comment (victorjulien) [reply src/app-layer-htp.c]: This table is used to match against strings set in the libhtp library, so we'll have to use what libhtp does.

**Comment by victorjulien** [comment]
https://redmine.openinfosecfoundation.org/issues/3186
https://redmine.openinfosecfoundation.org/issues/3187

File: src/app-layer-htp.c (hunk 1 above)
Comment (catenacyber) [reply src/app-layer-htp.c]: And this is a back port, so we should first fix master I guess...

**Pull request merged**